checkov/common/graph/graph_builder/graph_components/blocks.py

```diff
@@ -110,6 +110,14 @@ def get_attribute_dict(self, add_hash: bool = True) -> Dict[str, Any]:
             # since module names are user defined we are just setting 'module' as resource type for easier searching
             base_attributes[CustomAttributes.RESOURCE_TYPE] = "module"
 
+        if self.block_type == BlockType.PROVIDER:
+            provider_name = base_attributes[CustomAttributes.BLOCK_NAME]
+            if isinstance(provider_name, str):
+                provider_type = provider_name.split(".")[0]
+                base_attributes[CustomAttributes.RESOURCE_TYPE] = f"provider.{provider_type}"
+            else:
+                base_attributes[CustomAttributes.RESOURCE_TYPE] = f"provider.{base_attributes[CustomAttributes.BLOCK_NAME]}"
```

Comment:
```suggestion
        if self.block_type == BlockType.PROVIDER:
            # provider_name is always a string, base_attributes needs better typing 
            provider_name = cast(str, base_attributes[CustomAttributes.BLOCK_NAME])
            provider_type = provider_name.split(".")[0]
            base_attributes[CustomAttributes.RESOURCE_TYPE] = f"provider.{provider_type}"
```
and don't forget to import `cast` from `typing`